server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpMaxRcptHandlerTest.java

```diff
@@ -0,0 +1,106 @@
+import java.util.List;
+
+import org.apache.james.mailets.TemporaryJamesServer;
+import org.apache.james.mailets.configuration.SmtpConfiguration;
+import org.apache.james.modules.protocols.SmtpGuiceProbe;
+import org.apache.james.probe.DataProbe;
+import org.apache.james.utils.DataProbeImpl;
+import org.apache.james.utils.SMTPMessageSender;
+import org.apache.james.utils.SMTPSendingException;
+import org.apache.james.utils.SmtpSendingStep;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.api.io.TempDir;
+
+class SmtpMaxRcptHandlerTest {
+    private static final String USER = "user@" + DEFAULT_DOMAIN;
+    private static final Integer DEFAULT_MAX_RCPT_HANDLER = 50;
+
+    @RegisterExtension
+    public SMTPMessageSender messageSender = new SMTPMessageSender(DEFAULT_DOMAIN);
+
+    private TemporaryJamesServer jamesServer;
+
+    private void createJamesServer(File temporaryFolder, SmtpConfiguration.Builder smtpConfiguration) throws Exception {
+        jamesServer = TemporaryJamesServer.builder()
+            .withSmtpConfiguration(smtpConfiguration)
+            .build(temporaryFolder);
+        jamesServer.start();
+
+        DataProbe dataProbe = jamesServer.getProbe(DataProbeImpl.class);
+        dataProbe.addDomain(DEFAULT_DOMAIN);
+        dataProbe.addUser(USER, PASSWORD);
+
+        for (int i = 1; i <= DEFAULT_MAX_RCPT_HANDLER + 1 ; i++) {
+            dataProbe.addUser("recipient" + i + "@" + DEFAULT_DOMAIN, PASSWORD);
+        }
+    }
+
+    @AfterEach
+    void tearDown() {
+        if (jamesServer != null) {
+            jamesServer.shutdown();
+        }
+    }
+
+    @Test
+    void messageShouldNotBeAcceptedWhenMaxRcptHandlerExceeded(@TempDir File temporaryFolder) throws Exception {
+        createJamesServer(temporaryFolder, SmtpConfiguration.builder()
+            .doNotVerifyIdentity()
+            .withMaxRcptHandler(DEFAULT_MAX_RCPT_HANDLER));
+
+        assertThatThrownBy(() ->
+            messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+                .authenticate(USER, PASSWORD)
+                .sendMessageWithHeaders(USER, getRecipients(DEFAULT_MAX_RCPT_HANDLER + 1), "message"))
+            .isEqualTo(new SMTPSendingException(SmtpSendingStep.RCPT, "some error message\n"));
+    }
+
+    @Test
+    void messageShouldBeAcceptedWhenMaxRcptHandlerWithinLimit(@TempDir File temporaryFolder) throws Exception {
+        createJamesServer(temporaryFolder, SmtpConfiguration.builder()
+            .doNotVerifyIdentity()
+            .withMaxRcptHandler(DEFAULT_MAX_RCPT_HANDLER));
+
+        assertDoesNotThrow(() ->
+            messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+                .authenticate(USER, PASSWORD)
+                .sendMessageWithHeaders(USER, getRecipients(DEFAULT_MAX_RCPT_HANDLER - 1),"message"));
```

Comment:
```suggestion
                .sendMessageWithHeaders(USER, getRecipients(DEFAULT_MAX_RCPT_HANDLER),"message"));
```

Let's test exactly at the limit